lib/crawly_elixir_jobs/middleware/amazon_unique_request.ex

```diff
@@ -0,0 +1,36 @@
+defmodule CrawlyExlirJobs.MiddleWare.AmazonUniqueRequest do
+  # try to filter out request to same page on Amazon but does not entirely match existing request:
+  # ie "https://www.amazon.de/s?k=Drawing&page=3&qid=1598532609&ref=sr_pg_3" and "https://www.amazon.de/s?k=Drawing&page=3&qid=1598532611&ref=sr_pg_3"
+
+  require Logger
+
+  def run(request, state) do
+    unique_request_seen_requests = Map.get(state, :unique_request_seen_requests, [])
+
+    case Enum.any?(unique_request_seen_requests, &match_request(&1, request)) do
```

Comment:
This is going to slow us down quite a bit. The number of seen requests is going to be huge. I would prefer to visit several duplicates and to get more products. 

Another option is to avoid pages by regexp in URLs..

Reply:
I will try the regex suggestion :) I want to filter out as much duplicate as possible so I could also do more other requests to amazon before they don't like me anymore :D